src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -185,20 +184,27 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                         }
 
                         _currentConfiguration = configuration;
-
                     }
                     catch (Exception ex)
                     {
                         _fetchMetadataFailure = ex;
-                        _syncAfter = DateTimeUtil.Add(now.UtcDateTime, AutomaticRefreshInterval < RefreshInterval ? AutomaticRefreshInterval : RefreshInterval);
-                        if (_currentConfiguration == null) // Throw an exception if there's no configuration to return.
+
+                        if (_currentConfiguration == null)
+                        {
+                            _syncAfter = DateTimeUtil.Add(now.UtcDateTime, BootstrapRefreshInterval);
+
+                            // Throw an exception if there's no configuration to return.
                             throw LogHelper.LogExceptionMessage(
                                 new InvalidOperationException(
-                                    LogHelper.FormatInvariant(LogMessages.IDX20803, LogHelper.MarkAsNonPII(MetadataAddress ?? "null"), LogHelper.MarkAsNonPII(ex)), ex));
+                                    LogHelper.FormatInvariant(LogMessages.IDX20803, LogHelper.MarkAsNonPII(MetadataAddress ?? Utility.Null), LogHelper.MarkAsNonPII(ex)), ex));
+                        }
```

Comment:
This won't work. We can't do the `BootstrapRefreshInterval` for any exception it can only be if the IdP doesn't return any response (where we'll infer it never reached the IdP). If, e.g. the IdP simply returns metadata that can't be parse this code as written would DDOS the IdP potentially which isn't acceptable.

Reply:
This has been updated to retry up to 4 times with the BootstrapRefreshInterval, then fall back to the regular retry logic afterwards.